From 1663c7c2bcf475a379ae32ca21ce3753c8798279 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Thu, 11 Aug 2016 17:20:18 -0700
Subject: [PATCH 1730/2848] HADOOP-13154. S3AFileSystem
 printAmazonServiceException/printAmazonClientException
 appear copy & paste of AWS examples. Contributed
 by Steve Loughran.

(cherry picked from commit 0abb3663647cbadfb340328f623136060b82e1b7)

Change-Id: I139f2b0c3be5f47a00ef83e22364d1d8d4c01059
---
 .../org/apache/hadoop/fs/s3a/S3AFileSystem.java    |   25 ++++++--------------
 1 file changed, 7 insertions(+), 18 deletions(-)

diff --git a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
index c2052e6..fd6d817 100644
--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
@@ -1049,7 +1049,6 @@ public S3AFileStatus getFileStatus(Path f) throws IOException {
           throw e;
         }
       } catch (AmazonClientException e) {
-        printAmazonClientException(e);
         throw e;
       }
 
@@ -1079,7 +1078,6 @@ public S3AFileStatus getFileStatus(Path f) throws IOException {
             throw e;
           }
         } catch (AmazonClientException e) {
-          printAmazonClientException(e);
           throw e;
         }
       }
@@ -1125,7 +1123,6 @@ public S3AFileStatus getFileStatus(Path f) throws IOException {
         throw e;
       }
     } catch (AmazonClientException e) {
-      printAmazonClientException(e);
       throw e;
     }
 
@@ -1409,21 +1406,13 @@ public long getDefaultBlockSize() {
   }
 
   private void printAmazonServiceException(AmazonServiceException ase) {
-    LOG.info("Caught an AmazonServiceException, which means your request made it " +
-        "to Amazon S3, but was rejected with an error response for some reason.");
-    LOG.info("Error Message: " + ase.getMessage());
-    LOG.info("HTTP Status Code: " + ase.getStatusCode());
-    LOG.info("AWS Error Code: " + ase.getErrorCode());
-    LOG.info("Error Type: " + ase.getErrorType());
-    LOG.info("Request ID: " + ase.getRequestId());
-    LOG.info("Class Name: " + ase.getClass().getName());
-  }
-
-  private void printAmazonClientException(AmazonClientException ace) {
-    LOG.info("Caught an AmazonClientException, which means the client encountered " +
-        "a serious internal problem while trying to communicate with S3, " +
-        "such as not being able to access the network.");
-    LOG.info("Error Message: {}" + ace, ace);
+    LOG.info("Caught an AmazonServiceException {}", ase.toString());
+    LOG.info("Error Message: {}", ase.getMessage());
+    LOG.info("HTTP Status Code: {}", ase.getStatusCode());
+    LOG.info("AWS Error Code: {}", ase.getErrorCode());
+    LOG.info("Error Type: {}", ase.getErrorType());
+    LOG.info("Request ID: {}", ase.getRequestId());
+    LOG.info("Stack", ase);
   }
 
   /**
-- 
1.7.9.5

